### PR TITLE
feat: Ability define default Refresh mode on application level

### LINKED ISSUE #7774: app create with --annotations argocd.argoproj.io/refresh=hard`  annotations is not  effect
If you are trying to resolve an environment-specific issue or have a one-off question about the edge case that does not require a feature then please consider asking a question in argocd slack [channel](https://argoproj.github.io/community/join-slack).

Checklist:

* [√] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [√] I've included steps to reproduce the bug.
* [√] I've pasted the output of `argocd version`.

**Describe the bug**

`argocd app create ${app} --repo ${repo}  --helm-chart ${app} --revision ${revision} --dest-namespace ${namespace} --dest-server ${dest_server} --annotations argocd.argoproj.io/refresh=hard`
when use annotations  "argocd.argoproj.io/refresh=hard" but not take effect，
but use annotations  "refresh=hard" is effect

**To Reproduce**

`argocd app create nginx-test --repo http://nexus.sangfor.com/repository/test --helm-chart nginx-test --revision 2.0.9 --dest-namespace argo-app --dest-server https://kubernetes.default.svc --annotations argocd.argoproj.io/refresh=hard`

**Expected behavior**
> [root@master .kube]# kubectl get app/nginx-test -n argocd -o yaml
> apiVersion: argoproj.io/v1alpha1
> kind: Application
> metadata:
>   creationTimestamp: "2021-11-24T01:34:34Z"
>   generation: 10
>   name: nginx-test
>   namespace: argocd
>   resourceVersion: "27341226"
>   uid: 24d635dc-3aa1-47b6-8e04-904bea4c2699
>  spec:

but 

> [root@master .kube]# argocd app create nginx-test --repo http://nexus.sangfor.com/repository/test --helm-chart nginx-test --revision 2.0.9 --dest-namespace argo-app --dest-server https://kubernetes.default.svc --annotations refresh=hard
> application 'nginx-test' created
> [root@master .kube]# kubectl get app/nginx-test -n argocd -o yaml
> apiVersion: argoproj.io/v1alpha1
> kind: Application
> metadata:
>   **annotations:
>     refresh: hard**
>   creationTimestamp: "2021-11-24T01:38:15Z"
>   generation: 2
>   name: nginx-test
>   namespace: argocd
>   resourceVersion: "27343087"
>   uid: 12df8034-f84c-4f4a-941e-39fcd5401f99

**Version**
```shell
{
    "Version": "v2.1.7+a408e29"
    "kubectl":"v0.21.0"
}
```

### PR DESCRIPTION
**Pull request by pasha-codefresh**
Closes #7774

### CODE REVIEW

**Review by alexmt — CHANGES_REQUESTED** [review]
I think automatic hard refresh is useful in some cases but it is not safe to let end-user enable it.  

Argo CD performance depends on manifest caching a lot. Disabling caching for one application might affect other Argo CD tenants. Can we discuss this PR in contributors meeting please?

**Comment by itshaungmu** [comment]
1. This function is not enabled by default and does not affect the original cache. It feels very troublesome to trigger the hard refresh every time when deploying the application using helm. Turning on this function can facilitate automatic mandatory update detection of the helm application deployed in the development phase.
2. If you are worried that it may affect other tenants, can you only allow the admin to configure this parameter?

**Comment by pasha-codefresh** [comment]
@itshaungmu We have meeting today and I think we will conclude a solution. The problem that if you enable hard refresh even for the single application you can affect the whole cluster. I tend to agree with @alexmt comment. I will keep you updated

File: pkg/apis/application/v1alpha1/application_annotations.go [hunk 1]
```diff
@@ -5,6 +5,8 @@ const (
 	// Might take values 'normal'/'hard'. Value 'hard' means manifest cache and target cluster state cache should be invalidated before refresh.
 	AnnotationKeyRefresh string = "argocd.argoproj.io/refresh"
 
+	AnnotationKeyRefreshType string = "argocd.argoproj.io/refresh-type"
```
Comment (leoluz) [inline pkg/apis/application/v1alpha1/application_annotations.go:8]: Please add missing docs

**Comment by jessesuen** [comment]
I put my concern as well in this comment https://github.com/argoproj/argo-cd/issues/7774#issuecomment-985010716. This has come up before and we don't want to perform hard refreshes ever reconciliation.

**Pull request closed without merging**